src/main/java/com/cityelf/repository/UserRepository.java

```diff
@@ -77,35 +78,26 @@ public UserRepository() {
     return users;
   }
 
-  public User getUser(long id) {
-    for (User user : users) {
-      if (user.getId() == id) {
-        return user;
-      }
-    }
-    return null;
+  public Optional<User> getUser(long id) {
+    return users.stream().filter(user -> user.getId() == id).findFirst();
   }
 
   public void addNewUser(User user) {
     users.add(user);
   }
 
-  public void updateUser(User user) {
+  public boolean updateUser(User user) {
     long id = user.getId();
     for (User userDb : users) {
```

Comment:
You started writing code in functional style, so it would be better to use it everywhere. Change for cycle to **forEach** method